Sources/AardvarkReveal/RevealAttachmentGenerator.swift

```diff
@@ -0,0 +1,518 @@
+/// generator has completed bundling the Reveal file, user interaction can be restored.
+///
+/// # Security Settings
+///
+/// The generator communicates with the Reveal server over HTTP. By default, the App Transport Security policy requires
+/// the use of a secure connection, which is not available. To enable the generator to connect to the server, add the
+/// following to your app's `Info.plist`:
+///
+/// ```xml
+/// <key>NSAppTransportSecurity</key>
+/// <dict>
+///     <key>NSExceptionDomains</key>
+///     <dict>
+///         <key>localhost</key>
+///         <dict>
+///             <key>NSExceptionAllowsInsecureHTTPLoads</key>
+///             <true/>
+///             <key>NSIncludesSubdomains</key>
+///             <true/>
+///         </dict>
+///     </dict>
+/// </dict>
+/// ```
+@objc(ARKRevealAttachmentGenerator)
+public final class RevealAttachmentGenerator: NSObject {
+
+    // MARK: - Life Cycle
+
+    public override init() {
+        super.init()
+
+        serviceBrowser.startSearching()
+
+        notificationObservers.append(
+            NotificationCenter.default.addObserver(
+                forName: UIApplication.didEnterBackgroundNotification,
+                object: nil,
+                queue: .main
+            ) { [weak self] _ in
+                self?.serviceBrowser.stopSearching()
+            }
+        )
+
+        notificationObservers.append(
+            NotificationCenter.default.addObserver(
+                forName: UIApplication.willEnterForegroundNotification,
+                object: nil,
+                queue: .main
+            ) { [weak self] _ in
+                self?.serviceBrowser.startSearching()
+            }
+        )
+    }
+
+    deinit {
+        notificationObservers.forEach(NotificationCenter.default.removeObserver(_:))
+    }
+
+    // MARK: - Public Properties
+
+    public weak var delegate: RevealAttachmentGeneratorDelegate?
+
+    // MARK: - Private Properties
+
+    private let serviceBrowser: RevealServiceBrowser = .init()
+
+    private var notificationObservers: [NSObjectProtocol] = []
+
+    // MARK: - Public Methods
+
+    /// Begins the process of capturing the current app state and bundling the results into an attachment containing a
+    /// Reveal file.
+    ///
+    /// - parameter completionQueue: The queue on which to call the `completion`.
+    /// - parameter completion: The completion handler to call when the capture is complete. Called with `nil` when the
+    /// capture could not successfully be completed.
+    @available(iOS 11, *)
+    @objc
+    public func captureCurrentAppState(
+        completionQueue: DispatchQueue,
+        completion: @escaping (ARKBugReportAttachment?) -> Void
+    ) {
+        guard let service = serviceBrowser.localService else {
+            completionQueue.async {
+                completion(nil)
+            }
+            return
+        }
+
+        let port = service.port
+        guard port != -1 else {
```

Comment:
can we turn `-1` into a named constant?

Reply:
- [ ] Turn missing port check into named constant

Reply:
Addressed in 33d1f73